README.md
#Shoutbox
[![Software License](https://img.shields.io/badge/license-GPL-brightgreen.svg?style=flat)](LICENSE) 

Shoutbox is an XOOPS module that provides an interactive block and pop-up window for visitors to post and view messages.


docs/changelog.txt
<h5>5.02 Beta 2       [2017-07-11]</h5>
<hr>
- XOOPS 2.5.9, PHP 7 (mamba)
- removed @version (mamba)
- fixed issue #5: not working for custom group (myblackrx/mamba)
- Updates for PayPal changes (zyspec)
- PSR-2 code cosmetics (mamba)
- addNavigation(basename(__FILE__))  (mamba)
- dirname(__FILE__) to __DIR__ (mamba)
- replaced < br /> with < br > (mamba)
- Unnecessary double quotes (mamba)
- reference mismatch (mamba)
- Hardening to type safe (mamba)
- update Help (mamba)
- changed _handler to Handler (mamba)
- removed '/' from void HTML elements (&lt;br&gt;, &lt;img&gt;, etc) (mamba)
- added Contributing.md (mamba)
- renamed several $indexAdmin with $adminObject for consistency (mamba)
- added install checks for PHP 5.5 and XOOPS 2.5.8 (mamba)
- added auto-creation of upload folders (mamba)
- moved functions to Utility class (mamba)
- PHP 5 type casting (mamba)
- started conversion to XMF (mamba)
    &nbsp;&nbsp;- updated addInfoBoxLine entries (mamba)
    &nbsp;&nbsp;- displayNavigation(basename(__FILE__))  (mamba)
    &nbsp;&nbsp;- changed to ->displayButton('left') (mamba)

<h5>5.02 Beta 1       [2014-04-23]</h5>
<hr>
- moved all images, CSS, and JS files to /assets (mamba)
- renamed .html Smarty templates to .tpl (mamba)


<h5>5.01 Final  [2013-08-10]</h5>
<hr>
- cosmetic changes in Preferences (mamba)


<h5>5.01 Beta 1  [2012-07-30]</h5>
<hr>
- Converted to XOOPS 2.5.5 Admin GUI (Mamba)


<h5>5.00 Final  [2010-01-24]</h5>
<hr>
- Converted to Blue Move (Trabis)
- Classes were added and a lot of the code was rewritten.

Database has changed so please do not upload this package over the old shoutbox.
- Unistall shoutbox
- Delete all files
- Upload new files
- Install Shoutbox


<h5>4.05  [2008-12-15]</h5>
<hr>
- updates by Tank (Website: http://www.customvirtualdesigns.com)


<h5>Initial release 4.01  [10/05/2008]</h5>
<hr>
 - Posted messages can be stored in database table or csv file
 - All components of message entry are routed through sanitizer prior to storage
 - Dynamic resizing of pop-up window has been disabled until further testing
 - Text entry field now contains options for single line text box or
   multi-line text area with optional alerts when approaching message
   length limit
 - Display of registered user's avatar included in standard message block
 - OnUpdate function provided so database table will be created (if
   necessary) when updating from 4.0 beta

 NOTE: If your current of Shoutbox is earlier than v4.0 beta, it is
 not known whether performing a module update will work properly. Our
 recommendation would be to uninstall the current Shoutbox module and
 perform a New Install.

<h5>Update Release 4.02  [11/01/2008]</h5>
<hr>
 - Bug Fix: Typo in include/class.php causing some file storage processing
   to be performed incorrectly
 - Add guest avatar
 - Popup window dynamic resizing re-designed using div tags and css
 - Add captcha for guest posting with good guest mode
   The module contains it's own copy of XOOPS 2.3.0 captcha class with
   modifications to meet ShoutBox module requirements
 - Add captcha enable/disable preference parameter
 - Enable module style sheet so user can customize appearance
 - Change Refresh and Popup from links to buttons in Shoutbox block

<h5>Update Release 4.03  [11/15/2008]</h5>
<hr>
 - Bug Fix: Update text sanitizer functions to eliminate debug deprecated message
 - Add wordwrap preference parameter for controlling message line display length
 - Update style sheet to coincide with wordwrap feature so horizontal scroll bar
   is automatically removed when not necessary
 - Add display avatar preference parameter to enable/disable display of avatars
   in the ShoutBox block. NOTE: This has no affect on Popup window.
 - Remove local copy of captcha class.
 - Add captcha support for Frameworks/captcha class.

<h5>Update Release 4.04  [12/01/2008]</h5>
<hr>
 - Clean up some minor defects generating notices in debug mode
 - Bug Fix: Add passing UID to template to correct page refresh problem
 - Redefine how refresh works similar to the way it was in v4.02
   - Anytime an incorrect captcha code is entered the full page will refresh
     regardless of the type of captcha class used
   - If using Frameworks/captcha the full page will refresh for any message
     entry performed by guest
   - If using Core captcha only the message box, shout entry and captcha fields
     will be refreshed when an entry is performed by guest
 - Add word censoring support
 - Add selectable guest avatars. If guest avatar is set to None then blank.gif
   is placed in the avatar location.
 - Add option for disable/enable sound on new message when using popup window

<h5>Update Release 4.05  [12/15/2008]</h5>
<hr>
 - Add padding to sideboxcontent in module style sheet to provide some space
   between avatars and message text
 - Add javascript code in shoutframe and popupframe templates so URL's, when
   selected, open in a new window. (target ="_blank")
   NOTE: Special thanks to blueteen for this code
 - Add comments to shoutbox.css style sheet to help users in making changes
 - Add definitions in language files to eliminate all hard-coded text used
   in templates
 - Improve text sanitizing used in storing and retrieving messages
 - Eliminate extra line feeds in text entry when stored as File type

<h5>Credits:</h5>
<hr>
At the very beginning this module was developed for e-xoops aka runcms (XOOPS fork) but I ported it to XOOPS and never stopped porting... ;-)

Thx for everything...
- Herko [herkocoomans.net]: Where would I be without his spiritual help?
- Shine aka Triangle [adoptietrefpunt.nl]: Without her guidance I wouldn't have been the same person!
- All the rest I now forget but couldn't live without!

And those who helped developing me with shoutbox...
- Wizanda [wizanda.co.uk]: Kicking my ass for not following W3C guidelines... *shame*
- Masterprut [fueldump.de]: Helped me with those stupid javascript security measures (read: my scripting skills).

xoops_version.php
<?php
/*
 You may not change or alter any portion of this comment or credits
 of supporting developers from this source code or any supporting source code
 which is considered copyrighted (c) material of the original comment or credit authors.

 This program is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 */

/**
 * @copyright       The XUUPS Project http://sourceforge.net/projects/xuups/
 * @license         http://www.fsf.org/copyleft/gpl.html GNU public license
 * @package         Shoutbox
 * @author          Alphalogic <alphafake@hotmail.com>
 * @author          tank <tanksplace@comcast.net>
 * @author          trabis <lusopoemas@gmail.com>
 */

// defined('XOOPS_ROOT_PATH') || exit('XOOPS root path not defined');

$moduleDirName = basename(__DIR__);

// ------------------- Informations ------------------- //
$modversion = array(
    'version'             => '5.02',
    'module_status'       => 'Beta 2',
    'release_date'        => '2017/07/10',
    'name'                => _MI_SHOUTBOX_NAME,
    'description'         => _MI_SHOUTBOX_DESC,
    'credits'             => 'XOOPS Project',
    'help'                => 'page=>help',
    'license'             => 'GNU GPL 2.0 or later',
    'license_url'         => 'www.gnu.org/licenses/gpl-2.0.html',
    'official'            => 0, //1 indicates supported by XOOPS Dev Team, 0 means 3rd party supported
    'image'               => 'assets/images/logoModule.png',
    'dirname'             => basename(__DIR__),
    'author'              => 'Tank, Trabis',
    'modicons16'          => 'assets/images/icons/16',
    'modicons32'          => 'assets/images/icons/32',
    'module_website_url'  => 'www.xoops.org',
    'module_website_name' => 'XOOPS',
    'min_php'             => '5.5',
    'min_xoops'           => '2.5.8',
    'min_admin'           => '1.2',
    'min_db'              => array('mysql' => '5.5'),
    // Admin things
    'hasAdmin'            => 1,
    'system_menu'         => 1,
    'adminindex'          => 'admin/index.php',
    'adminmenu'           => 'admin/menu.php',
    // ------------------- Main Menu -------------------
    'hasMain'             => 1,
    // Sql file (must contain sql generated by phpMyAdmin or phpPgAdmin)
    'sqlfile'             => array('mysql' => 'sql/mysql.sql'),

    // Tables created by sql file (without prefix!)
    'tables'              => 'shoutbox',
);

// ------------------- Help files ------------------- //
$modversion['helpsection'] = array(
    ['name' => _MI_SHOUTBOX_OVERVIEW, 'link' => 'page=help'],
    ['name' => _MI_SHOUTBOX_DISCLAIMER, 'link' => 'page=disclaimer'],
    ['name' => _MI_SHOUTBOX_LICENSE, 'link' => 'page=license'],
    ['name' => _MI_SHOUTBOX_SUPPORT, 'link' => 'page=support'],
);

// ------------------- Blocks ------------------- //
$modversion['blocks'][] = array(
    'file'        => 'shoutbox.php',
    'name'        => _MI_SHOUTBOX_BLOCK,
    'description' => _MI_SHOUTBOX_DESC,
    'show_func'   => 'b_shoutbox_show',
    'edit_func'   => '',
    'template'    => 'shoutbox_block.tpl',
);


// ------------------- Templates ------------------- //
$modversion['templates'] = array(
    // User
    ['file' => 'shoutbox_popup.tpl', 'description' => 'Template for popup'],
    ['file' => 'shoutbox_shoutframe.tpl', 'description' => 'Template for block-iframe content'],
    ['file' => 'shoutbox_popupframe.tpl', 'description' => 'Template for popup-iframe content'],
    ['file' => 'shoutbox_popupheader.tpl', 'description' => ''],
    ['file' => 'shoutbox_online.tpl', 'description' => '']
);

//$modversion['onInstall']   = 'include/module.php';
//$modversion['onUninstall'] = 'include/module.php';

//Install/Uninstall Functions
$modversion['onInstall']   = 'include/oninstall.php';
//$modversion['onUpdate']    = 'include/onupdate.php';
$modversion['onUninstall'] = 'include/onuninstall.php';

// Config settings
// Global settings:
$modversion['config'][] = array(
    'name'        => 'logfile',
    'title'       => '_MI_SHOUTBOX_CAT1',
    'description' => '_MI_USERLOG_CONFCAT_LOGFILE_DSC',
    'formtype'    => 'line_break',
    'valuetype'   => 'textbox',
    'default'     => 'odd',
);
$modversion['config'][] = array(
    'name'        => 'guests_may_post',
    'title'       => '_MI_SHOUTBOX_TITLE1',
    'description' => '_MI_SHOUTBOX_EMPTY',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 1,
);
$modversion['config'][] = array(
    'name'        => 'guests_may_chname',
    'title'       => '_MI_SHOUTBOX_TITLE2',
    'description' => '_MI_SHOUTBOX_DESC2',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 0,
);
$modversion['config'][] = array(
    'name'        => 'allow_bbcode',
    'title'       => '_MI_SHOUTBOX_TITLE3',
    'description' => '_MI_SHOUTBOX_DESC3',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 1,
);
$modversion['config'][] = array(
    'name'        => 'stamp_format',
    'title'       => '_MI_SHOUTBOX_TITLE4',
    'description' => '_MI_SHOUTBOX_DESC4',
    'formtype'    => 'textbox',
    'valuetype'   => 'text',
    'default'     => 'h:i a',
);
$modversion['config'][] = array(
    'name'        => 'maxshouts_trim',
    'title'       => '_MI_SHOUTBOX_TITLE5',
    'description' => '_MI_SHOUTBOX_DESC5',
    'formtype'    => 'textbox',
    'valuetype'   => 'int',
    'default'     => '20',
);
$modversion['config'][] = array(
    'name'        => 'maxshouts_view',
    'title'       => '_MI_SHOUTBOX_TITLE6',
    'description' => '_MI_SHOUTBOX_DESC6',
    'formtype'    => 'textbox',
    'valuetype'   => 'int',
    'default'     => '20',
);
$modversion['config'][] = array(
    'name'        => 'storage_type',
    'title'       => '_MI_SHOUTBOX_TITLE7',
    'description' => '_MI_SHOUTBOX_DESC7',
    'formtype'    => 'select',
    'valuetype'   => 'text',
    'options'     => array('_MI_SHOUTBOX_OP7_F' => 'file', '_MI_SHOUTBOX_OP7_D' => 'database'),
    'default'     => 'file',
);
$modversion['config'][] = array(
    'name'        => 'user_realname',
    'title'       => '_MI_SHOUTBOX_TITLE8',
    'description' => '_MI_SHOUTBOX_DESC8',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 0,
);

// Block Settings:
$modversion['config'][] = array(
    'name'        => 'logfile',
    'title'       => '_MI_SHOUTBOX_CAT2',
    'description' => '_MI_USERLOG_CONFCAT_LOGFILE_DSC',
    'formtype'    => 'line_break',
    'valuetype'   => 'textbox',
    'default'     => 'odd',
);
$modversion['config'][] = array(
    'name'        => 'show_smileybar',
    'title'       => '_MI_SHOUTBOX_TITLE11',
    'description' => '_MI_SHOUTBOX_EMPTY',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 1,
);
$modversion['config'][] = array(
    'name'        => 'iframe_width',
    'title'       => '_MI_SHOUTBOX_TITLE12',
    'description' => '_MI_SHOUTBOX_DESC12',
    'formtype'    => 'textbox',
    'valuetype'   => 'text',
    'default'     => '100%',
);
$modversion['config'][] = array(
    'name'        => 'iframe_height',
    'title'       => '_MI_SHOUTBOX_TITLE13',
    'description' => '_MI_SHOUTBOX_DESC13',
    'formtype'    => 'textbox',
    'valuetype'   => 'text',
    'default'     => '150',
);
$modversion['config'][] = array(
    'name'        => 'iframe_border',
    'title'       => '_MI_SHOUTBOX_TITLE14',
    'description' => '_MI_SHOUTBOX_EMPTY',
    'formtype'    => 'textbox',
    'valuetype'   => 'text',
    'default'     => '0',
);
$modversion['config'][] = array(
    'name'        => 'popup',
    'title'       => '_MI_SHOUTBOX_TITLE15',
    'description' => '_MI_SHOUTBOX_DESC15',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 1,
);
$modversion['config'][] = array(
    'name'        => 'block_autorefresh',
    'title'       => '_MI_SHOUTBOX_TITLE16',
    'description' => '_MI_SHOUTBOX_DESC16',
    'formtype'    => 'select',
    'valuetype'   => 'int',
    'options'     => array('_MI_SHOUTBOX_OP16_BA0' => 0, '_MI_SHOUTBOX_OP16_BA1' => 1),
    'default'     => 1,
);
$modversion['config'][] = array(
    'name'        => 'wordwrap_setting',
    'title'       => '_MI_SHOUTBOX_TITLE17',
    'description' => '_MI_SHOUTBOX_DESC17',
    'formtype'    => 'text',
    'valuetype'   => 'int',
    'default'     => 0,
);
$modversion['config'][] = array(
    'name'        => 'display_avatar',
    'title'       => '_MI_SHOUTBOX_TITLE18',
    'description' => '_MI_SHOUTBOX_DESC18',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 1,
);
$modversion['config'][] = array(
    'name'        => 'guest_avatar',
    'title'       => '_MI_SHOUTBOX_TITLE19',
    'description' => '_MI_SHOUTBOX_DESC19',
    'formtype'    => 'select',
    'valuetype'   => 'int',
    'options'     => array(
        '_MI_SHOUTBOX_OP19_GA0' => 0,
        '_MI_SHOUTBOX_OP19_GA1' => 1,
        '_MI_SHOUTBOX_OP19_GA2' => 2,
        '_MI_SHOUTBOX_OP19_GA3' => 3,
        '_MI_SHOUTBOX_OP19_GA4' => 4,
        '_MI_SHOUTBOX_OP19_GA5' => 5
    ),
    'default'     => 1,
);
// PopUp:
$modversion['config'][] = array(
    'name'        => 'logfile',
    'title'       => '_MI_SHOUTBOX_CAT3',
    'description' => '_MI_USERLOG_CONFCAT_LOGFILE_DSC',
    'formtype'    => 'line_break',
    'valuetype'   => 'textbox',
    'default'     => 'odd',
);
$modversion['config'][] = array(
    'name'        => 'popup_whoisonline',
    'title'       => '_MI_SHOUTBOX_TITLE31',
    'description' => '_MI_SHOUTBOX_DESC31',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 1,
);
$modversion['config'][] = array(
    'name'        => 'popup_show_smileybar',
    'title'       => '_MI_SHOUTBOX_TITLE32',
    'description' => '_MI_SHOUTBOX_EMPTY',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 1,
);
$modversion['config'][] = array(
    'name'        => 'popup_sound',
    'title'       => '_MI_SHOUTBOX_TITLE33',
    'description' => '_MI_SHOUTBOX_EMPTY',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 1,
);
$modversion['config'][] = array(
    'name'        => 'popup_guests',
    'title'       => '_MI_SHOUTBOX_TITLE34',
    'description' => '_MI_SHOUTBOX_DESC34',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 0,
);
$modversion['config'][] = array(
    'name'        => 'popup_irc',
    'title'       => '_MI_SHOUTBOX_TITLE35',
    'description' => '_MI_SHOUTBOX_DESC35',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 1,
);
$modversion['config'][] = array(
    'name'        => 'popup_autofocus',
    'title'       => '_MI_SHOUTBOX_TITLE36',
    'description' => '_MI_SHOUTBOX_DESC36',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 1,
);
$modversion['config'][] = array(
    'name'        => 'popup_width',
    'title'       => '_MI_SHOUTBOX_TITLE37',
    'description' => '_MI_SHOUTBOX_DESC37',
    'formtype'    => 'textbox',
    'valuetype'   => 'text',
    'default'     => '500',
);
$modversion['config'][] = array(
    'name'        => 'popup_height',
    'title'       => '_MI_SHOUTBOX_TITLE38',
    'description' => '_MI_SHOUTBOX_DESC38',
    'formtype'    => 'textbox',
    'valuetype'   => 'text',
    'default'     => '340',
);

// Text Input:
$modversion['config'][] = array(
    'name'        => 'logfile',
    'title'       => '_MI_SHOUTBOX_CAT4',
    'description' => '_MI_USERLOG_CONFCAT_LOGFILE_DSC',
    'formtype'    => 'line_break',
    'valuetype'   => 'textbox',
    'default'     => 'odd',
);
$modversion['config'][] = array(
    'name'        => 'input_type',
    'title'       => '_MI_SHOUTBOX_TITLE40',
    'description' => '_MI_SHOUTBOX_DESC40',
    'formtype'    => 'select',
    'valuetype'   => 'int',
    'options'     => array('_MI_SHOUTBOX_OP40_TL' => 0, '_MI_SHOUTBOX_OP40_TA' => 1),
    'default'     => 1,
);
$modversion['config'][] = array(
    'name'        => 'textarea_rows',
    'title'       => '_MI_SHOUTBOX_TITLE41',
    'description' => '_MI_SHOUTBOX_DESC41',
    'formtype'    => 'textbox',
    'valuetype'   => 'text',
    'default'     => '4',
);
$modversion['config'][] = array(
    'name'        => 'textarea_cols',
    'title'       => '_MI_SHOUTBOX_TITLE42',
    'description' => '_MI_SHOUTBOX_DESC42',
    'formtype'    => 'textbox',
    'valuetype'   => 'text',
    'default'     => '75',
);
$modversion['config'][] = array(
    'name'        => 'text_linelength',
    'title'       => '_MI_SHOUTBOX_TITLE43',
    'description' => '_MI_SHOUTBOX_DESC43',
    'formtype'    => 'textbox',
    'valuetype'   => 'text',
    'default'     => '75',
);
$modversion['config'][] = array(
    'name'        => 'text_maxchars',
    'title'       => '_MI_SHOUTBOX_TITLE44',
    'description' => '_MI_SHOUTBOX_DESC44',
    'formtype'    => 'textbox',
    'valuetype'   => 'text',
    'default'     => '300',
);
$modversion['config'][] = array(
    'name'        => 'input_alerts',
    'title'       => '_MI_SHOUTBOX_TITLE45',
    'description' => '_MI_SHOUTBOX_DESC45',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 0,
);
$modversion['config'][] = array(
    'name'        => 'captcha_enable',
    'title'       => '_MI_SHOUTBOX_TITLE46',
    'description' => '_MI_SHOUTBOX_DESC46',
    'formtype'    => 'yesno',
    'valuetype'   => 'int',
    'default'     => 1,
);
